attribution_scopes.md

```diff
@@ -0,0 +1,229 @@
+  // Defaults to the empty list if omitted.
+  "attribution_scopes": <list of strings>,
+
+  // Optional
+  // Represents the maximum number of event states that an API caller plans to use across all
+  // subsequent event-source registrations.
+  // Example: default event source for event-level reports supports 1 attribution report, 1 reporting window,
+  // and 1 bit of trigger data for a total of 3 event states.
+  // This is used to calculate the information gain for event-level reports.
+  // Max event states must be integers > 0.
+  // Defaults to 3 if omitted.
+  // The flexible event-level script linked in the Privacy Considerations section below can be used to
+  // calculate number of states based on a configuration.
+  "max_event_states": <int>
+}
+```
+
+The following optional parameter will be added to the JSON in  `Attribution-Reporting-Register-Trigger` during trigger registration: `attribution_scopes`:
+
+```jsonc
+{
+  ..., // existing fields
+
+  // Optional
+  // Represents a list of attribution scopes for a particular trigger.
+  // Triggers will only match sources whose attribution_scopes contains at least one of the trigger's
+  // attribution_scopes, if specified.
+  // Attribution scope values must be strings. Each string has a maximum length of 50.
+  // Each list has a maximum length of 20.
+  // Defaults to the empty list if omitted.
+  "attribution_scopes": <list of strings>,
+}
+```
+
+Only sources whose `attribution_scopes` contains at least one of the trigger’s `attribution_scopes` will be considered for attribution. Note that this filtering takes place before attributing a trigger to a particular source.
+
+If there are multiple sources whose `attribution_scopes` contains at least one of the trigger’s `attribution_scopes`, then attribution will take place among the sources within the particular `attribution_scopes` according to current [Attribution Reporting API logic](https://github.com/WICG/attribution-reporting-api/blob/main/EVENT.md#trigger-attribution-algorithm).
+
+If the trigger registration's `attribution_scopes` is empty, then all sources are considered for attribution.
+
+Once an `attribution_scope_limit` is set, the last K values (where K = `attribution_scope_limit`) of `attribution_scopes` will be considered the final set of `attribution_scopes` values and any source with additional `attribution_scopes` values will be treated as if the attribution scopes were empty.
+
+If a source registration is specified with a configuration that has a higher number of event states than the most recent `max_event_states` for the same reporting origin, then the source will be rejected and the registration will fail. Additionally, if the `max_event_states` field is changed in a future source registration, then all other previous pending source registrations with a different `max_event_states` will be ignored in subsequent attribution report generation flows, but will still count towards rate limits. 
```

Comment:
Doesn't this whole paragraph only apply to sources that also specify attribution scope limit (AKA an "attribution-scope source")?